File: 2024_02/TP's/TP_1/palindromo.java
import java.util.*;

public class palindromo{
    public static void palindromo(String str){
        int result = 1; 
        int start = 0; 
        int end = str.length() - 1;

        while(start < end){
            if (str.charAt(start) != str.charAt(end)){
                result = 0;
                start = end;
            }
            start++;
            end--;
        }
        
        if(result == 1){
            System.out.println("SIM");
        } else{
            System.out.println("NAO");
        }
    }

    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        String line;
        
        line = sc.nextLine();
        
        while(!line.equals("FIM")){
                palindromo(line);
                line = sc.nextLine();
        } while(!line.equals("FIM"));
    }
}

File: 2024_02/TP's/TP_1/palindromo.c
#include <stdio.h>
#include <string.h>

int palindromo(char str[]){
    int result = 1; 
    int start = 0; 
    int end = strlen(str) - 1;

    while(start < end){
        if (str[start] != str[end]){
            result = 0;
            start = end;
        }
        start++;
        end--;
    }
    
    return result;
}

int main(void){
    char line[500];

    scanf("%[^\n]", line);

    while(strcmp(line, "FIM") != 0){
        if(palindromo(line) == 1){
            printf("SIM\n");
        } else{
            printf("NAO\n");
        }
        scanf(" %[^\n]", line);
    }

  return 0;
}
